File: src/leetcode/python/BinaryNumberWithAlternatingBits_693.py
class BinaryNumberWithAlternatingBits_693(object):
    def hasAlternatingBits(self, n):
        """
        :type n: int
        :rtype: bool
        """
        pm = -1
        while n:
            m = n & 1
            if pm == m:
                return False
            pm = m
            n >>= 1
        return True

File: src/leetcode/java/BinaryNumberWithAlternatingBits_693.java
package leetcode.java;

public class BinaryNumberWithAlternatingBits_693 {
    public boolean hasAlternatingBits(int n) {
        int pm = -1, m;
        while (n != 0) {
            m = n & 1;
            if (pm == m) return false;
            pm = m;
            n >>= 1;
        }
        return true;
    }
}
